### PR TITLE
runtime: removing correct_scheme_and_xfp

### PR DESCRIPTION
**Pull request by alyssawilk**
Fixes [#20977]

### CODE REVIEW

File: source/common/http/utility.h [hunk 1]
```diff
@@ -438,9 +438,7 @@ const std::string& getProtocolString(const Protocol p);
 
 /**
  * Return the scheme of the request.
- * For legacy code (envoy.reloadable_features.correct_scheme_and_xfp == false) this
- * will be the value of the X-Forwarded-Proto header value. By default it will
- * return the scheme if present, otherwise the value of X-Forwarded-Proto if
+ * By default it will return the scheme if present, otherwise the value of X-Forwarded-Proto if
```
Comment (adisuissa) [inline source/common/http/utility.h:441]: Will it still return X-Forwarded-Proto? Doesn't this PR changes it to only return the Scheme?

Or does this comment refer to this code:
https://github.com/envoyproxy/envoy/blob/a1f0d429fcd520497df40182adaac5efaaacb338/source/common/router/router.cc#L84-L91

**Review by adisuissa — COMMENTED** [review]
Thanks for the cleanup. Left a question about the comment.

File: source/common/http/utility.h (hunk 1 above)
Comment (alyssawilk) [reply source/common/http/utility.h:441]: yeah the prior comment was incorrect thanks.  I'll go remove the utility now that it's a wrapper around getSchemeValue

**Review by adisuissa — COMMENTED** [review]
LGTM thanks!
Please check CI.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**